rosbot_xl_gazebo/test/test_utils.py

```diff
@@ -15,95 +15,95 @@
 
 import rclpy
 from rclpy.node import Node
-from rclpy.executors import SingleThreadedExecutor
 
 from threading import Event
-from threading import Thread
 
 from geometry_msgs.msg import Twist
 from nav_msgs.msg import Odometry
-from sensor_msgs.msg import LaserScan, Image, PointCloud2
+from sensor_msgs.msg import LaserScan, Image, Imu, JointState, PointCloud2
 
 
-class SimulationTest:
+class SimulationTest(Node):
     __test__ = False
+
     # The inaccuracies in measurement uncertainties and wheel slippage
     # cause the rosbot_xl_base_controller to determine inaccurate odometry.
     ACCURACY = 0.10  # 10% accuracy
+    HARDWARE_PUBLISHERS_RATE = 10.0
+    VELOCITY_STABILIZATION_DELAY = 2
 
     def __init__(self, name="test_node", namespace=None):
-        self.VELOCITY_STABILIZATION_DELAY = 3
-        self.vel_stabilization_time_event = Event()
+        super().__init__(name, namespace=namespace)
+        self.cmd_vel_pub = self.create_publisher(Twist, "cmd_vel", 10)
+
+        # Robot callback
+        self.joint_sub = self.create_subscription(
+            JointState, "joint_states", self.joint_states_callback, 10
+        )
+        self.controller_sub = self.create_subscription(
+            Odometry, "rosbot_xl_base_controller/odom", self.controller_callback, 10
+        )
+        self.ekf_sub = self.create_subscription(
+            Odometry, "odometry/filtered", self.ekf_callback, 10
+        )
+        self.imu_sub = self.create_subscription(Imu, "imu_broadcaster/imu", self.imu_callback, 10)
+
+        # Sensor callback
+        self.camera_rgb_sub = self.create_subscription(
```

Comment:
Let's remove the sensors tests to another task. The sensors tests should depend on `camera_model` and `lidar_model`.